### DISCUSSION TITLE
After running production main.js file , get error: Uncaught ReferenceError: history is not defined

### DISCUSSION THREAD

**Discussion by andriiradkevych** [freshframework/fresh#2090] [General]
For building project to production , as I understand I have to execute `deno run -A dev.ts build` . This command generates me `_fresh` folder

<img width="223" alt="Screenshot 2023-11-29 at 01 57 10" src="https://github.com/denoland/fresh/assets/26725494/ade20f13-c3af-4480-ad5a-3bdebd75ea83">

In this folder I see `main.js` file , which as I understand I have to run . But when I run it by command `deno run -A _fresh/main.js` , I get on my termianl :

```
error: Uncaught ReferenceError: history is not defined
```
 Am I doing something wrong ? Could you pls give me more info how can make production build and run this file to handle my requests ?
 
 
 ```
 deno --version
 ```
```
deno 1.38.2 (release, aarch64-apple-darwin)
v8 12.0.267.1
typescript 5.2.2
```

**Comment by marvinhagemeister** [comment]
Ah I see where the confusion is coming from. The file in `_fresh/main.js` is the main entry for the browser, not for running it on the server. We should probably rename it to make the distinction more clear. To start the server for production use:

```diff
- deno run -A _fresh/main.js
+ deno run -A main.ts
```

Alternatively you can use the `deno task preview` command.
